pkg/gadgets/helpers.go

```diff
@@ -232,3 +233,57 @@ func FixBpfKtimeGetBootNs(programSpecs map[string]*ebpf.ProgramSpec) {
 		removeBpfKtimeGetBootNs(s)
 	}
 }
+
+// OptsWithMountnsMap adds an ebpf.Map to the ebpf.CollectionOptions and creates
+// the ebpf.CollectionOptions if it does not exist
+func OptsWithMountnsMap(opts *ebpf.CollectionOptions, mountnsMap *ebpf.Map) *ebpf.CollectionOptions {
+	if opts == nil {
+		opts = &ebpf.CollectionOptions{}
+	}
+	if opts.MapReplacements == nil {
+		opts.MapReplacements = map[string]*ebpf.Map{}
+	}
+	if mountnsMap != nil {
+		opts.MapReplacements[MntNsFilterMapName] = mountnsMap
+	}
+
+	return opts
+}
```

Comment:
I'm not sure this function is needed. I think we can put this code directly into `LoadeBPFSpec`

Reply:
✅